scripts/itest_parallel.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Get all the variables.
+PROCESSES=$1
+TRANCHES=$2
+TEST_FLAGS=$3
+ITEST_FLAGS=$4
+
+# Create a variable to hold the final exit code.
+exit_code=0
+
+# Run commands using xargs in parallel and capture their PIDs
+pids=()
+for ((i=0; i<PROCESSES; i++)); do 
+    cmd="scripts/itest_part.sh $i $TRANCHES $TEST_FLAGS $ITEST_FLAGS"
+    echo "$i" | $cmd & pids+=($!)
```

Comment:
Why not just:
```
for ((i=0; i<PROCESSES; i++)); do
  scripts/itest_part.sh $i "$TRANCHES" "$TEST_FLAGS" "$ITEST_FLAGS" &
  pids+=($!)
done
```

I'm not sure it's syntactically correct to have a second command following the `&` character (note, this is different from the `&&` _and_ operator and means: run that command in the background.
So I'm not sure if the rest of the line is interpreted correctly in all versions/environments.

Reply:
Nice! Even better!

Reply:
I would still add a line break after the `&`. Otherwise it's very easy to read it as `&&` if it is followed by another instruction.